internal/k8s/exec.go

```diff
@@ -170,14 +171,29 @@ func ReturnDeploymentObject(clientset *kubernetes.Clientset, matchLabel string,
 
 	log.Info().Msgf("waiting for %s Deployment to be created", matchLabelValue)
 
-	// Create watch operation
-	objWatch, err := clientset.
-		AppsV1().
-		Deployments(namespace).
-		Watch(context.Background(), deploymentListOptions)
-	if err != nil {
-		log.Error().Msgf("error when attempting to search for Deployment: %s", err)
-		return nil, err
+	var objWatch watch.Interface
+	var err error
+
+	timeout := 20 * time.Second
```

Comment:
To avoid you having to manually create timers, you can use context timers, like this:

```go
ctx, cancel := context.WithTimeout(20 * time.Second)
defer cancel()

for {
  select {
    case <-ctx.Done():
      // do something here, it means the timer expired, akin to your 
      // if time.Since(startTime) > timeout 
      
    default:
      // do whatever else you were supposed to do
  }
}


```